### PR TITLE
CC-12700 Introduce AvalaraTax module

### PR DESCRIPTION
**Pull request by AsonUnique**
- Jira ticket: https://spryker.atlassian.net/browse/CC-12700

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "spryker-eco/avalara-tax",
+  "description": "AvalaraTax module",
+  "license": "proprietary",
+  "require": {
+    "php": ">=7.3",
```
Comment (romanhavrylko) [inline composer.json:6]: Should `avalara-tax-extension` module be required?

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+     * Specification:
+     * - Version of the Avalara application.
+     *
+     * @api
+     */
+    public const AVALARA_TAX_APPLICATION_VERSION = 'AVALARA_TAX:APPLICATION_VERSION';
+
+    /**
+     * Specification:
+     * - The machine name of the machine on which this code is executing.
```
Comment (romanhavrylko) [inline src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php:33]: Please add `@api` tag

File: composer.json (hunk 1 above)
Comment (AsonUnique) [reply composer.json:6]: It will be added in the next PRs.

**Review by romanhavrylko — APPROVED** [review]

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @api
+     */
+    public const AVALARA_TAX_MACHINE_NAME = 'AVALARA_TAX:MACHINE_NAME';
+
+    /**
+     * Specification:
+     * - Indicates which server to use; acceptable values are "sandbox" or "production", or the full URL of your AvaTax instance.
```
Comment (abitskil) [inline src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: * `Indicates which server to use,`.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public const AVALARA_TAX_COMPANY_CODE = 'AVALARA_TAX:COMPANY_CODE';
+
+    /**
+     * Specification:
+     * - Commit a transaction for reporting after placing an order.
+     *
+     * @api
+     */
+    public const AVALARA_TAX_IS_TRANSACTION_COMMIT_AFTER_ORDER_PLACEMENT_ENABLED = 'AVALARA_TAX:IS_TRANSACTION_COMMIT_AFTER_ORDER_PLACEMENT_ENABLED';
```
Comment (abitskil) [inline src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: * This should be a module config.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+     * Specification:
+     * - Company identifier.
+     *
+     * @api
+     */
+    public const AVALARA_TAX_COMPANY_CODE = 'AVALARA_TAX:COMPANY_CODE';
+
+    /**
+     * Specification:
+     * - Commit a transaction for reporting after placing an order.
```
Comment (abitskil) [inline src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: * Commit the transaction for reporting after the order is placed.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+     * Specification:
+     * - Indicates which server to use; acceptable values are "sandbox" or "production", or the full URL of your AvaTax instance.
+     *
+     * @api
+     */
+    public const AVALARA_TAX_ENVIRONMENT_NAME = 'AVALARA_TAX:ENVIRONMENT_NAME';
+
+    /**
+     * Specification:
+     * - The account ID for your AvaTax account
```
Comment (abitskil) [inline src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: * Add the dot at the end.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @api
+     */
+    public const AVALARA_TAX_ACCOUNT_ID = 'AVALARA_TAX:ACCOUNT_ID';
+
+    /**
+     * Specification:
+     * - The private license key for your AvaTax account
```
Comment (abitskil) [inline src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: * Add the dot at the end.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        <column name="response" required="false" type="CLOB"/>
+
+        <behavior name="timestampable"/>
+
+        <id-method-parameter value="spy_avalara_tax_api_log_pk_seq"/>
+    </table>
+
+    <table name="spy_tax_avalara_sales_order">
+        <column name="id_tax_avalara_sales_order" required="true" type="INTEGER" primaryKey="true"
+                autoIncrement="true"/>
```
Comment (abitskil) [inline src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml]: * Remove the line break.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+        <column name="total_tax" type="DECIMAL" required="true" size="20" scale="10"/>
+        <column name="total_taxable" type="DECIMAL" required="true" size="20" scale="10"/>
+        <column name="total_tax_calculated" type="DECIMAL" required="true" size="20" scale="10"/>
+        <column name="adjustment_reason" type="VARCHAR" required="true" size="255"/>
+        <column name="adjustment_description" type="VARCHAR" required="false" size="255"/>
+        <column name="locked" type="BOOLEAN" required="true"/>
+        <column name="region" type="VARCHAR" required="true" size="255"/>
+        <column name="country" type="VARCHAR" required="true" size="255"/>
+        <column name="version" type="INTEGER" required="true"/>
+        <column name="softwareVersion" type="VARCHAR" required="true" size="255"/>
```
Comment (abitskil) [inline src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml]: * Rename to the `software_version`.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+        <behavior name="timestampable"/>
+
+        <id-method-parameter value="spy_tax_avalara_sales_order_pk_seq"/>
+    </table>
+
+    <table name="spy_tax_avalara_sales_order_item">
+        <column name="id_tax_avalara_sales_order_item" required="true" type="INTEGER" primaryKey="true"
+                autoIncrement="true"/>
+        <column name="fk_tax_avalara_sales_order" type="INTEGER" required="true"/>
+        <column name="fk_sales_order_item" type="INTEGER" required="false"/>
```
Comment (abitskil) [inline src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml]: * Mandatory field.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/AbstractSpyTaxAvalaraApiLog.php [hunk 11]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
```
Comment (abitskil) [inline src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/AbstractSpyTaxAvalaraApiLog.php:3]: * Switch to the MIT license header.
* Check other places for the same issue.

```suggestion
/**
 * MIT License
 * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
 */
```

File: composer.json [hunk 12]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "spryker-eco/avalara-tax",
+  "description": "AvalaraTax module",
+  "license": "proprietary",
```
Comment (abitskil) [inline composer.json]: * Switch to the `MIT`.

File: LICENSE [hunk 13]
```diff
@@ -0,0 +1,40 @@
+SPRYKER SYSTEMS GMBH EVALUATION LICENSE AGREEMENT
+
```
Comment (abitskil) [inline LICENSE]: * All `Eco` modules a published under the MIT license.
* Use as a source https://github.com/spryker-eco/oauth-azure/blob/master/LICENSE.

File: phpstan.json [hunk 14]
```diff
@@ -0,0 +1,3 @@
+{
+    "defaultLevel": 5
```
Comment (abitskil) [inline phpstan.json]: * Raise to the 8 level.

File: README.md [hunk 15]
```diff
@@ -0,0 +1,15 @@
+# AvalaraTax Module
+[![Build Status](https://travis-ci.org/spryker-eco/avalara-tax.svg)](https://travis-ci.org/spryker-eco/avalara-tax)
+[![Minimum PHP Version](https://img.shields.io/badge/php-%3E%3D%207.2-8892BF.svg)](https://php.net/)
+
+AvalaraTax module allows a Spryker project to connect to Avalara tax service.
```
Comment (abitskil) [inline README.md]: * AvalaraTax module allows the Spryker project to connect to the Avalara tax service.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php (hunk 3 above)
Comment (abitskil) [reply src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: Resolved.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php (hunk 4 above)
Comment (abitskil) [reply src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: Resolved.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php (hunk 5 above)
Comment (abitskil) [reply src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: Resolved.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php (hunk 6 above)
Comment (abitskil) [reply src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: Resolved.

File: src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php (hunk 7 above)
Comment (abitskil) [reply src/SprykerEco/Shared/AvalaraTax/AvalaraTaxConstants.php]: Resolved.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml (hunk 8 above)
Comment (abitskil) [reply src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml]: Resolved.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml (hunk 9 above)
Comment (abitskil) [reply src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml]: Resolved.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml (hunk 10 above)
Comment (abitskil) [reply src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/Schema/spy_avalara_tax.schema.xml]: Resolved.

File: src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/AbstractSpyTaxAvalaraApiLog.php (hunk 11 above)
Comment (abitskil) [reply src/SprykerEco/Zed/AvalaraTax/Persistence/Propel/AbstractSpyTaxAvalaraApiLog.php:3]: Resolved.

File: composer.json (hunk 12 above)
Comment (abitskil) [reply composer.json]: Resolved.

File: LICENSE (hunk 13 above)
Comment (abitskil) [reply LICENSE]: Resolved.

File: phpstan.json (hunk 14 above)
Comment (abitskil) [reply phpstan.json]: Resolved.

File: README.md (hunk 15 above)
Comment (abitskil) [reply README.md]: Resolved.

**Review by abitskil — APPROVED** [review]

**Pull request merged**